qadence_protocols/mitigations/readout.py

```diff
@@ -82,13 +107,58 @@ def renormalize_counts(corrected_counts: npt.NDArray, n_shots: int) -> npt.NDArr
 
         renormalization_factor = n_shots / sum_corrected_counts
         corrected_counts = np.rint(corrected_counts * renormalization_factor).astype(int)
+
+    # At this point, the count should be off by at most 2, added or substracted to/from the
+    # max count.
+    if sum(corrected_counts) != n_shots:
+        count_diff = sum(corrected_counts) - n_shots
+        max_count_bs = np.argmax(corrected_counts)
+        corrected_counts[max_count_bs] -= count_diff
+
     return corrected_counts
 
 
 def matrix_inv(K: npt.NDArray) -> npt.NDArray:
     return inv(K) if matrix_rank(K) == K.shape[0] else pinv(K)
 
 
+def constrained_inversion(noise_matrices: npt.NDArray, p_raw: npt.NDArray) -> npt.NDArray:
+    # Initial random guess in [0,1].
+    p_corr0 = np.random.rand(len(p_raw))
+    # Stochasticity constraints.
+    normality_constraint = LinearConstraint(np.ones(len(p_raw)).astype(int), lb=1.0, ub=1.0)
+    positivity_constraint = LinearConstraint(np.eye(len(p_raw)).astype(int), lb=0.0, ub=1.0)
+    constraints = [normality_constraint, positivity_constraint]
+    # Minimize the corrected probabilities.
+    res = minimize(corrected_probas, p_corr0, args=(noise_matrices, p_raw), constraints=constraints)
+
+    return res.x
+
+
+def majority_vote(noise_matrices: npt.NDArray, p_raw: npt.NDArray) -> npt.NDArray:
```

Comment:
It would have been preferable to do it in another PR for readability.

Reply:
Ye I started making some changes on it immediately after requesting the first PR. I will ask you about how to handle it in general in our next meeting